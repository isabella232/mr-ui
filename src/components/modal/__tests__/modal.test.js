import React from 'react';
import { shallow } from 'enzyme';
import { testCases } from './modal-test-cases';

describe('Modal', () => {
  let testCase;
  let wrapper;

  describe(testCases.basicSmall.description, () => {
    beforeEach(() => {
      testCase = testCases.basicSmall;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('fires onExit', () => {
      // find the button that triggers onExit method
      wrapper.find('IconButton').simulate('click');

      expect(testCase.props.onExit).toHaveBeenCalledTimes(1);
    });
  });

  describe(testCases.basicLarge.description, () => {
    beforeEach(() => {
      testCase = testCases.basicLarge;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('fires onExit', () => {
      // find the button that triggers onExit method
      wrapper.find('IconButton').simulate('click');

      expect(testCase.props.onExit).toHaveBeenCalledTimes(1);
    });
  });

  describe(testCases.basicAuto.description, () => {
    beforeEach(() => {
      testCase = testCases.basicAuto;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('fires onExit', () => {
      // find the button that triggers onExit method
      wrapper.find('IconButton').simulate('click');

      expect(testCase.props.onExit).toHaveBeenCalledTimes(1);
    });
  });

  describe(testCases.allOptions.description, () => {
    beforeEach(() => {
      testCase = testCases.allOptions;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe(testCases.optionalOnExit.description, () => {
    beforeEach(() => {
      testCase = testCases.optionalOnExit;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('IconButton is not present', () => {
      expect(wrapper.find('IconButton').exists()).toBe(false);
    });
  });
});
