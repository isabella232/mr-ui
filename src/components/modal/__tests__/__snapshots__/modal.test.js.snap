// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal No fixed width, display only renders 1`] = `
<Displaced
  dialogStyle={
    Object {
      "boxShadow": "none",
    }
  }
  focusDialog={true}
  getApplicationNode={[Function]}
  mounted={true}
  onExit={[MockFunction]}
  titleText="Auto modal title"
  underlayClass="bg-darken50 px12 py12 px60-mm py60-mm"
  underlayColor=""
  underlayProps={
    Object {
      "data-popover-ignore-clicks": true,
    }
  }
  underlayStyle={
    Object {
      "zIndex": 1,
    }
  }
>
  <div
    className="relative wmax-full  bg-white px36 py36 round"
  >
    <div>
      No fixed width on the modal container
    </div>
    <div
      className="absolute top right"
    >
      <IconButton
        icon="close"
        onClick={[MockFunction]}
        themeButton="btn btn--transparent unround-t unround-br color-gray py12 px12"
        tooltipProps={
          Object {
            "content": "Close",
          }
        }
      />
    </div>
  </div>
</Displaced>
`;

exports[`Modal all options renders 1`] = `
<Displaced
  dialogStyle={
    Object {
      "boxShadow": "none",
    }
  }
  getApplicationNode={[Function]}
  initialFocus="#foo"
  mounted={true}
  onExit={[MockFunction]}
  titleText="All options"
  underlayClass="bg-pink px24 py24"
  underlayColor=""
  underlayProps={
    Object {
      "data-foo": "foo",
    }
  }
  underlayStyle={
    Object {
      "margin": "pink",
      "zIndex": 1,
    }
  }
>
  <div
    className="relative wmax-full w360 bg-gray-light round py60 px60"
  >
    <div>
      <div
        className="mb12"
      >
        I am a message
      </div>
      <button
        className="btn"
        id="foo"
      >
        press me
      </button>
    </div>
    <div
      className="absolute top right"
    >
      <IconButton
        icon="close"
        onClick={[MockFunction]}
        themeButton="btn btn--transparent unround-t unround-br color-gray py12 px12"
        tooltipProps={
          Object {
            "content": "Close",
          }
        }
      />
    </div>
  </div>
</Displaced>
`;

exports[`Modal basic default renders 1`] = `
<Displaced
  dialogStyle={
    Object {
      "boxShadow": "none",
    }
  }
  focusDialog={true}
  getApplicationNode={[Function]}
  mounted={true}
  onExit={[MockFunction]}
  titleText="Large modal title"
  underlayClass="bg-darken50 px12 py12 px60-mm py60-mm"
  underlayColor=""
  underlayProps={
    Object {
      "data-popover-ignore-clicks": true,
    }
  }
  underlayStyle={
    Object {
      "zIndex": 1,
    }
  }
>
  <div
    className="relative wmax-full w600 bg-white px36 py36 round"
  >
    <div>
      Large modal body
    </div>
    <div
      className="absolute top right"
    >
      <IconButton
        icon="close"
        onClick={[MockFunction]}
        themeButton="btn btn--transparent unround-t unround-br color-gray py12 px12"
        tooltipProps={
          Object {
            "content": "Close",
          }
        }
      />
    </div>
  </div>
</Displaced>
`;

exports[`Modal basic small renders 1`] = `
<Displaced
  dialogStyle={
    Object {
      "boxShadow": "none",
    }
  }
  focusDialog={true}
  getApplicationNode={[Function]}
  mounted={true}
  onExit={[MockFunction]}
  titleText="Small modal title"
  underlayClass="bg-darken50 px12 py12 px60-mm py60-mm"
  underlayColor=""
  underlayProps={
    Object {
      "data-popover-ignore-clicks": true,
    }
  }
  underlayStyle={
    Object {
      "zIndex": 1,
    }
  }
>
  <div
    className="relative wmax-full w360 bg-white px36 py36 round"
  >
    <div>
      Small modal body
    </div>
    <div
      className="absolute top right"
    >
      <IconButton
        icon="close"
        onClick={[MockFunction]}
        themeButton="btn btn--transparent unround-t unround-br color-gray py12 px12"
        tooltipProps={
          Object {
            "content": "Close",
          }
        }
      />
    </div>
  </div>
</Displaced>
`;

exports[`Modal no onExit renders 1`] = `
<Displaced
  dialogStyle={
    Object {
      "boxShadow": "none",
    }
  }
  focusDialog={true}
  getApplicationNode={[Function]}
  mounted={true}
  titleText="No onExit passed as prop"
  underlayClass="bg-darken50 px12 py12 px60-mm py60-mm"
  underlayColor=""
  underlayProps={
    Object {
      "data-popover-ignore-clicks": true,
    }
  }
  underlayStyle={
    Object {
      "zIndex": 1,
    }
  }
>
  <div
    className="relative wmax-full w360 bg-white px36 py36 round"
  >
    <div>
      You can't close me
    </div>
  </div>
</Displaced>
`;
